diff --git a/utils/airtime-silan b/utils/airtime-silan
index 3bffb0ee2..19f429c3a 100755
--- a/utils/airtime-silan
+++ b/utils/airtime-silan
@@ -49,7 +49,7 @@ try:
             full_path = f['fp']
             # silence detect(set default queue in and out)
             try:
-                command = ['silan', '-b' '-f', 'JSON', full_path]
+                command = ['silan', '-b', '-f', 'JSON', full_path]
                 proc = subprocess.Popen(command, stdout=subprocess.PIPE)
                 out = proc.communicate()[0].strip('\r\n')
                 info = json.loads(out)
