diff --git a/airtime_mvc/application/models/StoredFile.php b/airtime_mvc/application/models/StoredFile.php
index 2845b7a69..0c1287aa4 100644
--- a/airtime_mvc/application/models/StoredFile.php
+++ b/airtime_mvc/application/models/StoredFile.php
@@ -513,6 +513,10 @@ SQL;
     	$serverPort = $_SERVER['SERVER_PORT'];
     	$subDir = $CC_CONFIG['baseDir'];
 
+        if ($protocol === 'https' && $serverPort == 80) {
+            $serverPort = 443;
+        }
+
     	if ($subDir[0] === "/") {
     		$subDir = substr($subDir, 1, strlen($subDir) - 1);
