diff --git a/python_apps/api_clients/api_clients/api_client.py b/python_apps/api_clients/api_clients/api_client.py
index 6710c95..db9f8d7 100644
--- a/python_apps/api_clients/api_clients/api_client.py
+++ b/python_apps/api_clients/api_clients/api_client.py
@@ -185,8 +185,9 @@ class RequestProvider(object):
         self.requests = {}
         if self.config["general"]["base_dir"].startswith("/"):
             self.config["general"]["base_dir"] = self.config["general"]["base_dir"][1:]
-        self.url = ApcUrl("http://%s:%s/%s%s/%s" \
-            % (self.config["general"]["base_url"], str(self.config["general"]["base_port"]),
+        self.url = ApcUrl("%s://%s:%s/%s%s/%s" \
+            % (str(("https", "http")[self.config["general"]["base_port"] == 443]),
+               self.config["general"]["base_url"], str(self.config["general"]["base_port"]),
                self.config["general"]["base_dir"], self.config["api_base"],
                '%%action%%'))
         # Now we must discover the possible actions
@@ -322,8 +323,9 @@ class AirtimeApiClient(object):
         # TODO : Make other methods in this class use this this method.
         if self.config["general"]["base_dir"].startswith("/"):
             self.config["general"]["base_dir"] = self.config["general"]["base_dir"][1:]
-        url = "http://%s:%s/%s%s/%s" %  \
-            (self.config["general"]["base_url"], str(self.config["general"]["base_port"]),
+        url = "%s://%s:%s/%s%s/%s" %  \
+            (str(("https", "http")[self.config["general"]["base_port"] == 443]),
+             self.config["general"]["base_url"], str(self.config["general"]["base_port"]),
              self.config["general"]["base_dir"], self.config["api_base"],
              self.config[config_action_key])
         url = url.replace("%%api_key%%", self.config["general"]["api_key"])
