diff --git a/python_apps/pypo/liquidsoap/generate_liquidsoap_cfg.py b/python_apps/pypo/liquidsoap/generate_liquidsoap_cfg.py
index d8eb275..6dbb37f 100644
--- a/python_apps/pypo/liquidsoap/generate_liquidsoap_cfg.py
+++ b/python_apps/pypo/liquidsoap/generate_liquidsoap_cfg.py
@@ -25,7 +25,6 @@ def generate_liquidsoap_config(ss):
         fh.write(str_buffer.encode('utf-8'))
 
     auth_path = os.path.dirname(os.path.realpath(__file__))
-    fh.write('log_file = "/var/log/airtime/pypo-liquidsoap/<script>.log"\n')
     fh.write('auth_path = "%s/liquidsoap_auth.py"\n' % auth_path)
     fh.close()
     
diff --git a/python_apps/pypo/liquidsoap/liquidsoap.cfg b/python_apps/pypo/liquidsoap/liquidsoap.cfg
index d76e3af..d73ec49 100644
--- a/python_apps/pypo/liquidsoap/liquidsoap.cfg
+++ b/python_apps/pypo/liquidsoap/liquidsoap.cfg
@@ -26,7 +26,6 @@ s3_bitrate = 160
 ###########################################
 # Logging settings                        #
 ###########################################
-log_file = "/var/log/airtime/pypo-liquidsoap/<script>.log"
 #log_level = 3
 
 ###########################################
diff --git a/python_apps/pypo/liquidsoap/ls_script.liq b/python_apps/pypo/liquidsoap/ls_script.liq
index 632cf63..778262b 100644
--- a/python_apps/pypo/liquidsoap/ls_script.liq
+++ b/python_apps/pypo/liquidsoap/ls_script.liq
@@ -1,6 +1,7 @@
 %include "/etc/airtime/liquidsoap.cfg"
 
-set("log.file.path", log_file)
+set("log.file", false)
+set("log.stdout", true)
 set("server.telnet", true)
 set("server.telnet.port", 1234)
 # set("init.daemon.pidfile.path", "/var/run/airtime/airtime-liquidsoap.pid")
