diff --git a/python_apps/media-monitor/mm2/media/monitor/pure.py b/python_apps/media-monitor/mm2/media/monitor/pure.py
index fc6692b..67724a0 100644
--- a/python_apps/media-monitor/mm2/media/monitor/pure.py
+++ b/python_apps/media-monitor/mm2/media/monitor/pure.py
@@ -259,7 +259,7 @@ def organized_path(old_path, root_path, orig_md):
     filepath = None
     ext = extension(old_path)
     def default_f(dictionary, key):
-        if key in dictionary: return len(str(dictionary[key])) == 0
+        if key in dictionary: return len(dictionary[key].encode('ascii', 'ignore')) == 0
         else: return True
     # We set some metadata elements to a default "unknown" value because we use
     # these fields to create a path hence they cannot be empty Here "normal"
