diff --git a/python_apps/media-monitor/mm2/media/monitor/manager.py b/python_apps/media-monitor/mm2/media/monitor/manager.py
index c457b39..c4c8c05 100644
--- a/python_apps/media-monitor/mm2/media/monitor/manager.py
+++ b/python_apps/media-monitor/mm2/media/monitor/manager.py
@@ -187,7 +187,8 @@ class Manager(Loggable):
             try: mmp.create_dir(path)
             except mmp.FailedToCreateDir as e: self.unexpected_exception(e)
 
-        os.chmod(store_paths['organize'], 0775) 
+        # RaBe: dont' blindly chmod things, thats up to the admin
+        #os.chmod(store_paths['organize'], 0775)
 
         self.set_problem_files_path(store_paths['problem_files'])
         self.set_imported_path(store_paths['imported'])
