diff --git a/python_apps/pypo/setup.py b/python_apps/pypo/setup.py
index 76011e7..f59f201 100644
--- a/python_apps/pypo/setup.py
+++ b/python_apps/pypo/setup.py
@@ -17,19 +17,7 @@ else:
         for filename in filenames:
             pypo_files.append(os.path.join(root, filename))
         
-    data_files = [
-                  ('/etc/init', ['install/upstart/airtime-playout.conf.template']),
-                  ('/etc/init', ['install/upstart/airtime-liquidsoap.conf.template']),
-                  ('/etc/init.d', ['install/sysvinit/airtime-playout']),
-                  ('/etc/init.d', ['install/sysvinit/airtime-liquidsoap']),
-                  ('/var/log/airtime/pypo', []),
-                  ('/var/log/airtime/pypo-liquidsoap', []),
-                  ('/var/tmp/airtime/pypo', []),
-                  ('/var/tmp/airtime/pypo/cache', []),
-                  ('/var/tmp/airtime/pypo/files', []),
-                  ('/var/tmp/airtime/pypo/tmp', []),
-                 ]
-    print data_files
+    data_files = []
 
 setup(name='airtime-playout',
       version='1.0',
