diff --git a/python_apps/pypo/setup.py b/python_apps/pypo/setup.py
index 76011e7..a88fb4d 100644
--- a/python_apps/pypo/setup.py
+++ b/python_apps/pypo/setup.py
@@ -17,19 +17,7 @@ else:
         for filename in filenames:
             pypo_files.append(os.path.join(root, filename))
         
-    data_files = [
-                  ('/etc/init', ['install/upstart/airtime-playout.conf.template']),
-                  ('/etc/init', ['install/upstart/airtime-liquidsoap.conf.template']),
-                  ('/etc/init.d', ['install/sysvinit/airtime-playout']),
-                  ('/etc/init.d', ['install/sysvinit/airtime-liquidsoap']),
-                  ('/var/log/airtime/pypo', []),
-                  ('/var/log/airtime/pypo-liquidsoap', []),
-                  ('/var/tmp/airtime/pypo', []),
-                  ('/var/tmp/airtime/pypo/cache', []),
-                  ('/var/tmp/airtime/pypo/files', []),
-                  ('/var/tmp/airtime/pypo/tmp', []),
-                 ]
-    print data_files
+    data_files = []
 
 setup(name='airtime-playout',
       version='1.0',
@@ -48,7 +36,6 @@ setup(name='airtime-playout',
       install_requires=[
           'amqplib',
           'anyjson',
-          'argparse',
           'configobj',
           'docopt',
           'kombu',
@@ -57,8 +44,7 @@ setup(name='airtime-playout',
           'PyDispatcher',
           'pyinotify',
           'pytz',
-          'requests',
-          'wsgiref'
+          'requests'
       ],
       zip_safe=False,
       data_files=data_files)
